src/main/java/core/basesyntax/service/operation/ReturnOperationHandler.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.StorageService;
+
+public class ReturnOperationHandler implements OperationHandler {
+    private final StorageService storageService;
+
+    public ReturnOperationHandler(StorageService storageService) {
+        this.storageService = storageService;
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction, StorageService storageService) {
+        storageService.addFruit(transaction.getFruit(),transaction.getQuantity());
```

Comment:
The `handle` method implementation is correct, but be sure to validate the input data. For example, you should check if the `transaction` and its fields (`fruit` and `quantity`) are not null to prevent `NullPointerExceptions`. It's part of defensive programming to ensure data integrity.